code/modules/mod/modules/modules_general.dm

```diff
@@ -655,3 +655,108 @@
+	icon_state = "hat_holder"
+	module_type = MODULE_PASSIVE //The description says it auto-deploys for certain hats; that's just an excuse to have a whitelist of items.
+	complexity = 0 //Its FREE!
+	active_power_cost = 0 //Yes, thats right, FREE!
+	removable = FALSE
+	///Currently "stored" hat. No armor or function will be inherited, ONLY the icon.
+	var/obj/item/clothing/head/attached_hat
+	var/attachable_hats_list = list(/obj/item/clothing/head/caphat,)
+
+/obj/item/mod/module/hat_stabilizer/on_equip()
+	RegisterSignal(mod.helmet, COMSIG_PARENT_EXAMINE, .proc/add_examine)
+	RegisterSignal(mod.helmet, COMSIG_PARENT_ATTACKBY, .proc/place_hat)
+	RegisterSignal(mod.helmet, COMSIG_ATOM_ATTACK_HAND_SECONDARY, .proc/remove_hat)
+	attachable_hats_list = typecacheof(
+		//List of attachable hats. Make sure these and their subtypes are all tested, so they dont appear janky.
+		//This list should also be gimmicky, so captains can have fun. I.E. the Santahat, Pirate hat, Tophat, Chefhat...
+		//Yes, I said it, the captian should have fun.
+		list(
+			/obj/item/clothing/head/caphat,
+			/obj/item/clothing/head/crown,
+			/obj/item/clothing/head/centhat,
+			/obj/item/clothing/head/centcom_cap,
+			/obj/item/clothing/head/pirate,
+			/obj/item/clothing/head/santa,
+			/obj/item/clothing/head/hardhat/reindeer,
+			/obj/item/clothing/head/sombrero,
+			/obj/item/clothing/head/kitty,
+			/obj/item/clothing/head/rabbitears,
+			/obj/item/clothing/head/festive,
+			/obj/item/clothing/head/powdered_wig,
+			/obj/item/clothing/head/weddingveil,
+			/obj/item/clothing/head/that,
+			/obj/item/clothing/head/nursehat,
+			/obj/item/clothing/head/chefhat,
+			)) - /obj/item/clothing/head/caphat/beret
+			//Need to subtract the beret because its annoying
+
+/obj/item/mod/module/hat_stabilizer/on_unequip()
+	if(attached_hat)	//knock off the helmet if its on their head. Or, technically, auto-rightclick it for them; that way it saves us code, AND gives them the bubble
+		remove_hat()
+	attachable_hats_list = null
+	mod.helmet.desc = initial(mod.helmet.desc)	//Reset this, as our add_examine proc wont be firing any more
+	UnregisterSignal(mod.helmet, COMSIG_PARENT_EXAMINE)
+	UnregisterSignal(mod.helmet, COMSIG_PARENT_ATTACKBY)
+	UnregisterSignal(mod.helmet, COMSIG_ATOM_ATTACK_HAND_SECONDARY)
+
+/obj/item/mod/module/hat_stabilizer/on_suit_deactivation()
+	if(attached_hat)	//knock off the helmet if its on their head. Or, technically, auto-rightclick it for them; that way it saves us code, AND gives them the bubble
+		remove_hat(src, usr)
+
+/obj/item/mod/module/hat_stabilizer/proc/add_examine()
+	SIGNAL_HANDLER
+	mod.helmet.desc = initial(mod.helmet.desc)	//Reset this to prevent repeatedly adding notices.
+	if(attached_hat)
+		mod.helmet.desc += span_notice("\nThere's [attached_hat.name] placed on the helmet. Right-click to remove it.")
+	else
+		mod.helmet.desc += span_notice("\nThere's nothing placed on the helmet. Yet.")
```

Comment:
thats now how signal examines work, just do
. += your stuff

Reply:
ohhh, that explains why it was so janky; I'll test it, `. += "text"`, if I understood this right

Reply:
[was not fixed by bigfix commit. In fact, was broken by it.]